File: project/app/src/main/java/org/owntracks/android/ui/preferences/editor/EditorMvvm.java
package org.owntracks.android.ui.preferences.editor;
import androidx.databinding.Bindable;

import org.owntracks.android.ui.base.view.MvvmView;
import org.owntracks.android.ui.base.viewmodel.MvvmViewModel;
public interface EditorMvvm {

    interface View extends MvvmView {
        void displayLoadFailed();

        void displayExportToFileFailed();
        void displayExportToFileSuccessful();

        boolean exportConfigurationToFile(String exportStr);
    }

    interface ViewModel<V extends MvvmView> extends MvvmViewModel<V> {
        void onPreferencesValueForKeySetSuccessful();
        @Bindable String getEffectiveConfiguration();
    }
}

File: project/app/src/main/java/org/owntracks/android/ui/preferences/editor/EditorMvvm.kt
package org.owntracks.android.ui.preferences.editor

import androidx.databinding.Bindable
import org.owntracks.android.ui.base.view.MvvmView
import org.owntracks.android.ui.base.viewmodel.MvvmViewModel

interface EditorMvvm {
    interface View : MvvmView {
        fun displayLoadFailed()
        fun exportConfigurationToFile(): Boolean
    }

    interface ViewModel<V : MvvmView?> : MvvmViewModel<V> {
        fun onPreferencesValueForKeySetSuccessful()

        @get:Bindable
        val effectiveConfiguration: String?
    }
}
